### PR TITLE
Describe Agama's startup process

### PR DESCRIPTION
**Pull request by imobachgs**
* Describe the startup process.
* Update the README.md removing some outdated information.

### CODE REVIEW

File: doc/startup.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+## Web UI
+
+When discussing the web UI, we can distinguish two sides: the server process and the web browser.
+Regarding the server, Agama's web UI is implemented as a Cockpit module, so the only requisite is
+that the `cockpit.socket` is enabled. Then, you can connect to the UI using the
+`https://$SERVER_IP:9090/cockpit/@localhost/agama/index.html`.
+
+When using Agama Live, a local web browser is automatically started. In the default image, it is
+launched using an Icewm startup script.
```
Comment (jreidinger) [inline doc/startup.md:40]: any link to that startup script?

**Review by jreidinger — APPROVED** [review]
just 1 NP

File: doc/startup.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+## Auto-installation
+
+If the `agama.auto` option is specified in the kernel command-line, the
+[agama-auto.service](../service/share/systemd/agama-auto.service) comes into play. It runs after the
+`agama.service` so the D-Bus daemon is ready and the services can be activated as needed.
+
+## Web UI
+
+When discussing the web UI, we can distinguish two sides: the server process and the web browser.
+Regarding the server, Agama's web UI is implemented as a Cockpit module, so the only requisite is
```
Comment (mchf) [inline doc/startup.md:35]: i'd say ```prerequisite``` or ```requirement```

**Comment by mchf** [comment]
also just one minor comment. LGTM otherwise

File: doc/startup.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
+# Agama Startup Process
+
+This document summarizes how Agama's startup process.
+
+## Overview
+
+As described in the [README](../README.md#architecture), Agama is composed by a set of D-Bus
```
Comment (mvidner) [inline doc/startup.md:7]:
```suggestion
As described in the [README](../README.md#architecture), Agama is composed of a set of D-Bus
```

File: doc/startup.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+services up and running and make the web UI available. Additionally, the auto-installation procedure
+could be started if required by the user.
+
+The startup process is handled by systemd and D-Bus. The only exception is starting the local
+browser in the Agama Live image.
+
+## Starting the D-Bus Services
+
+[agama.service](../service/share/agama.service) is responsible for starting up Agama's D-Bus daemon
+process using the `agamactl --daemon` command. This process uses a [specific
```
Comment (mvidner) [inline doc/startup.md:18]:
```suggestion
process using the `agamactl --daemon` command. This process uses a dedicated bus with a [specific
```

File: doc/startup.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
+# Agama Startup Process
+
+This document summarizes how Agama's startup process.
```
Comment (mvidner) [inline doc/startup.md:3]:
```suggestion
This document summarizes Agama's startup process.
```

Comment (mvidner) [reply doc/startup.md:3]: Probably off topic, related to what was mentioned on the daily: do we want to mention what happens _before_ Agama starts up? Like, "see Linuxrc and Iguana..."

Comment (imobachgs) [reply doc/startup.md:3]: Well, that's more related to agama-live itself. But I will try to come up with something when we have the ALP-based image working.

**Review by mvidner — APPROVED** [review]

**Pull request merged**